src/library-authoring/create-collection/CreateCollectionModal.tsx

```diff
@@ -0,0 +1,131 @@
+  const {
+    isCreateCollectionModalOpen,
+    closeCreateCollectionModal,
+  } = React.useContext(LibraryContext);
+  const { showToast } = React.useContext(ToastContext);
+
+  const [isCreatingCollection, setIsCreatingCollection] = React.useState<boolean>(false);
+
+  const handleOnClose = React.useCallback(() => {
+    closeCreateCollectionModal();
+    setIsCreatingCollection(false);
+  }, []);
+
+  const handleCreate = React.useCallback((values) => {
+    setIsCreatingCollection(true);
+
+    create.mutateAsync(values).then(() => {
+      handleOnClose();
+      showToast(intl.formatMessage(messages.createCollectionSuccess));
+    }).catch(() => {
+      setIsCreatingCollection(false);
+      showToast(intl.formatMessage(messages.createCollectionError));
+    });
+  }, []);
+
+  return (
+    <ModalDialog
+      title={intl.formatMessage(messages.createCollectionModalTitle)}
+      isOpen={isCreateCollectionModalOpen}
+      onClose={handleOnClose}
+      size="xl"
+      hasCloseButton
+      isFullscreenOnMobile
+    >
+      <ModalDialog.Header>
+        <ModalDialog.Title>
+          {intl.formatMessage(messages.createCollectionModalTitle)}
+        </ModalDialog.Title>
+      </ModalDialog.Header>
+
+      <Formik
+        initialValues={{
+          title: '',
+          description: '',
+        }}
+        validationSchema={
+          Yup.object().shape({
+            title: Yup.string()
+              .required(intl.formatMessage(messages.createCollectionModalNameInvalid)),
+            description: Yup.string(),
+          })
+        }
+        onSubmit={(values) => handleCreate(values)}
+      >
+        {(formikProps) => (
+          <>
+            <ModalDialog.Body className="mw-sm">
+              <Form onSubmit={formikProps.handleSubmit}>
+                <FormikControl
+                  name="title"
+                  label={(
+                    <Form.Label className="font-weight-bold h3">
+                      {intl.formatMessage(messages.createCollectionModalNameLabel)}
+                    </Form.Label>
+                  )}
+                  value={formikProps.values.title}
+                  placeholder={intl.formatMessage(messages.createCollectionModalNamePlaceholder)}
+                  help=""
+                  className=""
+                  controlClasses="pb-2"
+                />
+                <FormikControl
+                  name="description"
+                  as="textarea"
+                  label={(
+                    <Form.Label className="font-weight-bold h3">
+                      {intl.formatMessage(messages.createCollectionModalDescriptionLabel)}
+                    </Form.Label>
+                  )}
+                  value={formikProps.values.description}
+                  placeholder={intl.formatMessage(messages.createCollectionModalDescriptionPlaceholder)}
+                  help={intl.formatMessage(messages.createCollectionModalDescriptionDetails)}
```

Comment:
This generates a warning, but the problem is in our `FormikControl` wrapper.

You can fix the `label` and `help` types in the `FormikControl.jsx` like the following
![image](https://github.com/user-attachments/assets/ff75b188-ee19-4c66-a059-27ede778e24b)

Or circumvent it by passing an element:
```
                  help={(<>{intl.formatMessage(messages.createCollectionModalDescriptionDetails)}</>)}
```

Reply:
Updated: https://github.com/openedx/frontend-app-course-authoring/pull/1259/commits/636d2312451f3c3745591292de2ffd510eb40294